### PR TITLE
[#954] update terraform to 0.12.* version

### PR DESCRIPTION
**Pull request by karbyshevds**
Upgrade to Terraform 12

### CODE REVIEW

**Review by alexander-semenets — APPROVED** [review]

File: terraform/modules/k8s/base_setup/main.tf [hunk 1]
```diff
@@ -1,40 +1,41 @@
...
-    name        = "${var.cluster_name}-tls"
-    namespace   = "${element(var.tls_namespaces, count.index)}"
+    name      = "${var.cluster_name}-tls"
+    namespace = element(var.tls_namespaces, count.index)
   }
-  data {
-    "tls.key"   = "${data.aws_s3_bucket_object.tls-secret-key.body}}"
-    "tls.crt"   = "${data.aws_s3_bucket_object.tls-secret-crt.body}}"
+  data = {
+    "tls.key" = "${data.aws_s3_bucket_object.tls-secret-key.body}}"
```
Comment (mcdoker18) [inline terraform/modules/k8s/base_setup/main.tf]: Please, fix variable value

File: terraform/modules/k8s/dashboard/main.tf [hunk 2]
```diff
@@ -4,49 +4,50 @@ provider "helm" {
...
-    namespace   = "kube-system"
+    name      = var.dashboard_tls_secret_name
+    namespace = "kube-system"
   }
-  data {
-    "tls.key"   = "${var.tls_secret_key}"
-    "tls.crt"   = "${var.tls_secret_crt}}"
+  data = {
+    "tls.key" = var.tls_secret_key
+    "tls.crt" = "${var.tls_secret_crt}}"
```
Comment (mcdoker18) [inline terraform/modules/k8s/dashboard/main.tf]: the same

File: terraform/modules/k8s/monitoring/main.tf [hunk 3]
```diff
@@ -4,63 +4,64 @@ provider "helm" {
...
-    namespace   = "${var.monitoring_namespace}"
+    name      = "${var.cluster_name}-tls"
+    namespace = var.monitoring_namespace
   }
-  data {
-    "tls.key"   = "${var.tls_secret_key}"
-    "tls.crt"   = "${var.tls_secret_crt}}"
+  data = {
+    "tls.key" = var.tls_secret_key
+    "tls.crt" = "${var.tls_secret_crt}}"
```
Comment (mcdoker18) [inline terraform/modules/k8s/monitoring/main.tf]: the same

File: terraform/modules/legion/main.tf [hunk 4]
```diff
@@ -11,130 +11,130 @@ provider "helm" {
...
-    name        = "${var.cluster_name}-tls"
-    namespace   = "${var.legion_namespace}"
+    name      = "${var.cluster_name}-tls"
+    namespace = var.legion_namespace
   }
-  data {
-    "tls.key"   = "${data.aws_s3_bucket_object.tls-secret-key.body}}"
-    "tls.crt"   = "${data.aws_s3_bucket_object.tls-secret-crt.body}}"
+  data = {
+    "tls.key" = "${data.aws_s3_bucket_object.tls-secret-key.body}}"
```
Comment (mcdoker18) [inline terraform/modules/legion/main.tf]: the same

**Review by mcdoker18 — CHANGES_REQUESTED** [review]

File: terraform/modules/k8s/base_setup/main.tf (hunk 1 above)
Comment (karbyshevds) [reply terraform/modules/k8s/base_setup/main.tf]: done in f3c7217

File: terraform/modules/legion/main.tf (hunk 4 above)
Comment (karbyshevds) [reply terraform/modules/legion/main.tf]: fixed in c789669

File: terraform/modules/k8s/monitoring/main.tf (hunk 3 above)
Comment (karbyshevds) [reply terraform/modules/k8s/monitoring/main.tf]: fixed in c789669

File: terraform/modules/k8s/dashboard/main.tf (hunk 2 above)
Comment (karbyshevds) [reply terraform/modules/k8s/dashboard/main.tf]: fixed in c789669

**Review by mcdoker18 — APPROVED** [review]

**Pull request merged**